Branch-SDK/src/test/java/io/branch/referral/BranchUtilTest.java

```diff
@@ -0,0 +1,102 @@
+package io.branch.referral;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class BranchUtilTest {
+
+    private final int[] resourceIds = new int[]{
+            0,
+            1,
+            32,
+            323429,
+            Integer.MAX_VALUE,
+            -1,
+            -50,
+            -476063,
+            Integer.MIN_VALUE
+    };
+
+    private final String[] encodedIds = new String[]{
+            "resourceID 0x0",
+            "resourceID 0x1",
+            "resourceID 0x20",
+            "resourceID 0x4ef65",
+            "resourceID 0x7fffffff",
+            "resourceID 0xffffffff",
+            "resourceID 0xffffffce",
+            "resourceID 0xfff8bc61",
+            "resourceID 0x80000000"
+    };
+
+    private final String[] replacePositiveJsonTests = new String[] {
+            //positive test cases
+            "{ \"field\": 1 }",
+            "{ \"field\": \"two\" }",
+            "{ \"field\": [] }",
+            "{ \"field\": [ 1, 2, \"three\", 4.0 ] }",
+            "{ \"field\": { \"field\": 5 } }",
+    };
+
+    private final String[] replaceNegativeJsonTests = new String[] {
+            "{}",
+            "{ \"replaced\": 0 }",
+            "{ \"field\": 1, \"replaced\": 2 }",
+            "{ \"field\": \"two\", \"replaced\": \"four\" }",
+            "{ \"field\": [], \"replaced\": [] }",
+            "{ \"field\": [ 1, 2, \"three\", 4.0 ], \"replaced\": [ 5, 6, \"seven\", 8.0 ] }",
+            "{ \"field\": { \"field\": 5 }, \"replaced\": { \"field\": 6 } }"
+    };
+
+    @Test
+    public void testEncodeResourceId() {
+        for (int i = 0; i < resourceIds.length; i++) {
+            Assert.assertEquals(encodedIds[i], BranchUtil.encodeResourceId(resourceIds[i]));
+        }
+    }
+
+    @Test
+    public void testDecodeResourceId() {
+        Assert.assertEquals(resourceIds[6], BranchUtil.decodeResourceId(encodedIds[6]));
```

Comment:
?

Reply:
good catch, will remove. (was debugging that test case).